symfony/plugins/orangehrmLeavePlugin/Service/LeaveAssignmentService.php

```diff
@@ -19,101 +19,120 @@
+use OrangeHRM\Entity\Employee;
+use OrangeHRM\Entity\Leave;
+use OrangeHRM\Entity\LeaveRequest;
+use OrangeHRM\Entity\LeaveRequestComment;
+use OrangeHRM\Entity\LeaveStatus;
+use OrangeHRM\Entity\WorkflowStateMachine;
 use OrangeHRM\Leave\Dto\LeaveParameterObject;
+use OrangeHRM\Leave\Exception\LeaveAllocationServiceException;
+use OrangeHRM\Leave\Traits\Service\LeaveEntitlementServiceTrait;
 
-class LeaveAssignmentService extends AbstractLeaveAllocationService {
+class LeaveAssignmentService extends AbstractLeaveAllocationService
+{
 
     protected $leaveEntitlementService;
+    use LeaveEntitlementServiceTrait;
+    use EntityManagerHelperTrait;
+    use AuthUserTrait;
+
     protected $dispatcher;
-    protected $assignWorkflowItem;
+    protected $assignWorkflowItem = null;
 
     /**
      * Get LeaveEntitlementService
      * @return LeaveEntitlementService
-     * 
+     *
      */
-    public function getLeaveEntitlementService() {
+    public function getLeaveEntitlementService()
+    {
+        // TODO
         if (!($this->leaveEntitlementService instanceof LeaveEntitlementService)) {
             $this->leaveEntitlementService = new LeaveEntitlementService();
         }
         return $this->leaveEntitlementService;
     }
-    
+
     /**
      * Set LeaveEntitlementService
-     * @param LeaveEntitlementService $service 
+     * @param LeaveEntitlementService $service
      */
-    public function setLeaveEntitlementService(LeaveEntitlementService $service) {
+    public function setLeaveEntitlementService(LeaveEntitlementService $service)
+    {
+        // TODO
         $this->leaveEntitlementService = $service;
     }
-    
+
     /**
      * Set dispatcher.
-     * 
+     *
      * @param $dispatcher
      */
-    public function setDispatcher($dispatcher) {
+    public function setDispatcher($dispatcher)
+    {
+        // TODO
         $this->dispatcher = $dispatcher;
     }
 
-    public function getDispatcher() {
-        if(is_null($this->dispatcher)) {
+    public function getDispatcher()
+    {
+        // TODO
+        if (is_null($this->dispatcher)) {
             $this->dispatcher = sfContext::getInstance()->getEventDispatcher();
         }
         return $this->dispatcher;
-    }  
-    
+    }
+
     /**
-     *
-     * @param array $leaveAssignmentData
-     * @return bool
+     * @param LeaveParameterObject $leaveAssignmentData
+     * @return LeaveRequest|null
+     * @throws LeaveAllocationServiceException
      */
-    public function assignLeave(LeaveParameterObject $leaveAssignmentData) {
-
+    public function assignLeave(LeaveParameterObject $leaveAssignmentData)
+    {
         $employeeId = $leaveAssignmentData->getEmployeeNumber();
-
         /* Check whether employee exists */
         if (empty($employeeId)) {
-            throw new LeaveAllocationServiceException('Invalid Employee');
+            throw LeaveAllocationServiceException::invalidEmployeeSelected();
         }
```

Comment:
will remove this, since `$leaveAssignmentData->getEmployeeNumber()` non-nullable. remove `invalidEmployeeSelected` from `LeaveAllocationServiceException`

Reply:
Done